### PR TITLE
feat: add hotreloading for portable experiences

### PR DESCRIPTION
**Pull request by leanmendoza**
temp: add hotreloading for portable experiences

# What? 
...

# Why? 
...

### CODE REVIEW

File: packages/entryPoints/index.ts [hunk 1]
```diff
@@ -320,13 +330,51 @@ export async function startPreview() {
...
+                'main',
+                wearable.name,
+                `${wearable.baseUrl}/`,
+                wearable.data.scene,
+                wearable.thumbnail
+              )
+              sceneLoading.set(wearable, false)
+            }
+          }
+        })()
```
Comment (menduz) [inline packages/entryPoints/index.ts:374]: Please catch errors here and print to the stderr

File: packages/unity-interface/dcl.ts [hunk 2]
```diff
@@ -179,14 +182,51 @@ export async function getPreviewSceneId(): Promise<{ sceneId: string | null; sce
...
-  if (sceneId) {
-    await reloadScene(sceneId)
+export async function loadPreviewScene({ ws, message }: { ws?: string; message?: any }) {
+  if (
+    message &&
+    message.payload &&
+    message.payload.body?.sceneType === 'portable-experience' &&
+    message.payload.body?.sceneId
+  ) {
+    ;(async () => {
```
Comment (gonpombo8) [inline packages/unity-interface/dcl.ts:192]: is this anonymous fn necessary ?

File: packages/unity-interface/dcl.ts [hunk 3]
```diff
@@ -179,14 +182,51 @@ export async function getPreviewSceneId(): Promise<{ sceneId: string | null; sce
...
+        const wearable = collection.data[0]
+        if (!sceneLoading.get(wearable.id)) {
+          await killPortableExperienceScene(wearable.id)
+
+          sceneLoading.set(wearable, true)
+          // This timeout is because the killPortableExperience isn't really async
+          //  and before spawn the portable experience it's neccesary that be kill
+          //  the previous scene
+          // TODO: catch the Scene.unloaded and then call the spawn.
+          await new Promise((resolve) => setTimeout(resolve, 100))
```
Comment (gonpombo8) [inline packages/unity-interface/dcl.ts:207]: im pretty sure that we have a sleep fn at some place in this repo.
If now can we create it for future uses ?

File: packages/unity-interface/dcl.ts [hunk 4]
```diff
@@ -179,14 +182,51 @@ export async function getPreviewSceneId(): Promise<{ sceneId: string | null; sce
...
+    message.payload &&
+    message.payload.body?.sceneType === 'portable-experience' &&
+    message.payload.body?.sceneId
+  ) {
+    ;(async () => {
+      const sceneId = message.payload.body?.sceneId
+      const url = `${rootURLPreviewMode()}/preview-wearables/${sceneId}`
+      const collection: { data: any[] } = await (await fetch(url)).json()
+
+      if (collection.data.length > 0) {
```
Comment (gonpombo8) [inline packages/unity-interface/dcl.ts:197]:
```ts
      if (!!collection.data.length) {
```

File: packages/entryPoints/index.ts [hunk 5]
```diff
@@ -326,7 +327,7 @@ export async function startPreview() {
         defaultLogger.info('Message received: ', message)
       }
 
-      loadPreviewScene(wsScene)
+      loadPreviewScene({ ws: wsScene, message })
```
Comment (gonpombo8) [inline packages/entryPoints/index.ts:330]: 🎉

File: packages/shared/wearablesPortableExperience/sagas.ts [hunk 6]
```diff
@@ -109,13 +109,18 @@ function* handleStartWearablesPortableExperience(action: StartWearablesPortableE
...
-      const gameJs = wearableContent.filter((c) => c.key.endsWith('game.js'))[0].key
-      const gameJsRootLength = gameJs.lastIndexOf('/') + 1
-
-      // Convert wearable mapping to scene's mapping and modify file path so game.js directory becomes the root directory
-      // for asset loading
-      const mappings = wearableContent.map((c) => ({ file: c.key.substring(gameJsRootLength), hash: c.hash }))
+      const femaleCrop = wearableContent.filter(($) => $.key.substr(0, 7) == 'female/').length == wearableContent.length
+      const maleCrop = wearableContent.filter(($) => $.key.substr(0, 5) == 'male/').length == wearableContent.length
+
+      let mappings: any = []
```
Comment (gonpombo8) [inline packages/shared/wearablesPortableExperience/sagas.ts:115]: can we add some comments to that female/male crop ?
Also if the uniq thing that changes in the if/else clause its the file prop can we move it to a fn ?

```ts
const getFile = (key: string) => {
  if (femaleCrop) return key.substring(7)
  if (maleCrop) return key.subString(5)
  return key
}
const mappings = wearableContent.map(
  content => ({ file: getFile(content.file), hash: content.hash })
)
```

This way mapping will have types, and also inmutable 😄

**Review by gonpombo8 — CHANGES_REQUESTED** [review]

**Review by gonpombo8 — APPROVED** [review]

**Review by gonpombo8 — APPROVED** [review]

**Pull request merged**